torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -212,9 +232,12 @@ def checkpoint(module: nn.Module, *, use_reentrant: bool = True) -> nn.Module:
     """
     torch._C._log_api_usage_once("torch.distributed.checkpoint")
 
-    def forward_pre_hook(module: nn.Module, inputs: Tuple[Any, ...]) -> None:
+    def forward_pre_hook(
+        module: nn.Module, args: Tuple[Any, ...], kwargs: Dict[str, Any]
+    ) -> None:
         if checkpoint.state(module).enable_hook:
             checkpoint.state(module).orig_grad_enabled = torch.is_grad_enabled()
+            flat_args, kwarg_keys = _pack_kwargs(*args, **kwargs)
```

Comment:
neat, I didn't know distributed had this utility.

Reply:
I checked the implementation and it doesn't use pytrees. Would that be an improvement? It would allow you to handle things like nested lists.

Reply:
(none of this has to happen this PR, or ever, just asking questions)